ams/server/src/main/java/com/netease/arctic/server/optimizing/plan/AbstractPartitionPlan.java

```diff
@@ -302,4 +304,67 @@ public TaskDescriptor buildTask(OptimizingInputProperties properties) {
           partition, input, properties.getProperties());
     }
   }
+
+  /**
+   * util class for bin-pack
+   */
+  protected static class FileTask {
+    private final IcebergDataFile file;
+    private final List<IcebergContentFile<?>> deleteFiles;
+    private final boolean isFragment;
+
+    public FileTask(IcebergDataFile file, List<IcebergContentFile<?>> deleteFiles, boolean isFragment) {
+      this.file = file;
+      this.deleteFiles = deleteFiles;
+      this.isFragment = isFragment;
+    }
+
+    public IcebergDataFile getFile() {
+      return file;
+    }
+
+    public List<IcebergContentFile<?>> getDeleteFiles() {
+      return deleteFiles;
+    }
+
+    public boolean isFragment() {
+      return isFragment;
+    }
+
+    public boolean isSegment() {
+      return !isFragment;
+    }
+  }
+
+  protected class BinPackingTaskSplitter implements TaskSplitter {
+
+    @Override
+    public List<SplitTask> splitTasks(int targetTaskCount) {
```

Comment:
I think we should use the "targetTaskCount" parameter to control how to split. If we split according to "targetFileSize", the merged file size will definitely be smaller than or equal to "targetFileSize", which is not what we expected.

Reply:
@wangtaohz  Can you reply to this question?

Reply:
Using `targetFileSize` to split tasks has this flaw, which is an old issue inherited from 0.4.1 (PR #821). I believe that making the split task size `n` times larger than `targetFileSize` can solve this problem.

As for `targetTaskCount`, currently this parameter cannot guarantee that tasks are split into reasonable sizes, more work needs to be done in the generation of `targetTaskCount` to ensure that it is used reasonably for splitting.